File: naruto-backend/src/main/java/naruto_backend/model/Ninja.java
package naruto_backend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;

@Entity
public class Ninja extends Utilisateur {

	@OneToMany
    @JoinColumn(name="equipe")
	@Column(name = "equipe", nullable = false)
	private Equipe equipe;

	public Ninja() {
	}

	public Ninja(Integer id, String login, String password, String prenom, String nom, Genre genre,
			Specialite specialite, RangNinja rang, EtatNinja etatNinja, int niveau, int expActuel, int pvMax,
			int pvActuel, int chakraMax, int chakraActuel, int nbEchecs, int nbReussites, int argent) {
		super(id, login, password, prenom, nom, genre, specialite, rang, etatNinja, niveau, expActuel, pvMax, pvActuel,
				chakraMax, chakraActuel, nbEchecs, nbReussites, argent);
	}

	public Equipe getEquipe() {
		return equipe;
	}

	public void setEquipe(Equipe equipe) {
		this.equipe = equipe;
	}

	@Override
	public String toString() {
		return "Ninja [equipe=" + equipe + "]";
	}

}

File: naruto-frontend/src/app/model/ninja.ts
import { Genre } from "./genre";
import { Utilisateur } from "./utilisateur";

export class Ninja extends Utilisateur {
  constructor (
      id: number,
      login: string,
      password: string,
      nom: string,
      prenom: string,
      genre: Genre,
      niveau: number,
      experience: number,
      pvMax: number,
      pvActuels: number,
      chakraMax: number,
      chakraActuel: number,
      nbEchecs: number,
      nbReussites: number,
      argent: number,
    public equipeId: number
  ) {
    super(id, login, password, nom, prenom, genre, niveau, experience, pvMax, pvActuels, chakraMax, chakraActuel, nbEchecs, nbReussites, argent)
  }
}
